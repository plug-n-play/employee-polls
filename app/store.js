import { configureStore } from "@reduxjs/toolkit";

import user from "@/features/user/userSlice";
import questions from "@/features/questions/questionsSlice";
import leaderboard from "@/features/leaderboard/leaderboardSlice";

export function makeStore() {
  return configureStore({
    reducer: {
      user,
      questions,
      leaderboard,
    }
  });
}

const store = makeStore();

export default store;
