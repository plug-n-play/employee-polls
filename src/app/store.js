import { configureStore } from "@reduxjs/toolkit";

import user from "@/features/user/userSlice";
import questions from "@/features/questions/questionsSlice";

export function makeStore() {
  return configureStore({
    reducer: {
      user,
      questions,
    }
  });
}

const store = makeStore();

export default store;
