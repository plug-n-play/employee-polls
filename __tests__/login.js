import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { makeStore } from '@/app/store';
import Login from '@/pages/login';

test('renders login page', async () => {
  const store = makeStore();
  const { getByText, getByTestId } = render(
    <Provider store={store}>
      <Login />
    </Provider>
  );

  expect(getByText(/Login/i)).toBeInTheDocument();
  expect(render).toMatchSnapshot();

  const usernameEl = getByTestId("username").querySelector('input');
  const passwordEl = getByTestId("password").querySelector('input');

  fireEvent.change(usernameEl, { target: { value: 'tylermcginnis' } });
  fireEvent.change(passwordEl, { target: { value: 'xxxxxxxxxx' } });

  const submitButton = document.querySelector('button');
  fireEvent.click(submitButton);
  expect(getByText(/Login/i)).toBeInTheDocument();
});