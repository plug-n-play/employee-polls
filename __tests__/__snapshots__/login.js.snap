// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login page 1`] = `[Function]`;
